app/services/case_filter/case_partial_case_flag_filter.rb

```diff
@@ -0,0 +1,51 @@
+module CaseFilter
+  class CasePartialCaseFlagFilter < CaseMultiChoicesFilterBase
+
+    def self.identifier
+      'filter_partial_case_flag'
+    end
+  
+    def self.filter_attributes
+      [:filter_partial_case_flag]
+    end
+
+    def available_choices
+      {
+        :filter_partial_case_flag => {
+          'partial-case' => I18n.t('filters.filter_partial_case_flag.partial-case'),
+          'not-partial-case'     => I18n.t('filters.filter_partial_case_flag.not-partial-case')
+        }
+      }
+    end
+
+    def is_permitted_for_user?
+      @user.permitted_correspondence_types.any? { | c_type | ['OFFENDER_SAR'].include? c_type.abbreviation }
+    end
+
+    def call
+      records = @records
+      records = filter_partial_case(records)
+      records
+    end
+
+    private
+
+    def is_partial_case?
+      'partial-case'.in? @query.filter_partial_case_flag
+    end
+
+    def is_not_partial_case?
+      'not-partial-case'.in? @query.filter_partial_case_flag
+    end
+
+    def filter_partial_case(records)
+      if is_partial_case? && !is_not_partial_case?
```

Comment:
Why the double check here isn't it that `is_partial_case?` is equivalent to `!is_not_partial_case?` ?

Reply:
sorry for this confusion,  the "is_partial_case" and "is_not_partial_case" represents the options from the filter, if only put 'is_partial_case', then 2 situation will fall in this categories 
  - user chose 'is_partial_case' only
  - user chose 'is_partial_case' and also 'is_not_partial_case'
then for first one, it will return partial cases only, but it would be wrong for the second situation,  it should return all cases, not only partial case.   So I put two to make sure the second situation won't be in this category. 

Save reason for your next comment

Reply:
👍 thanks for confirming :)